core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -171,21 +189,28 @@ class ControllerChannelManager(controllerEpoch: () => Int,
       case Some(name) => s"$name:Controller-${config.brokerId}-to-broker-${broker.id}-send-thread"
     }
 
+    val brokerMetricTag = brokerMetricTags(broker.id)
     val requestRateAndQueueTimeMetrics = metricsGroup.newTimer(
-      RequestRateAndQueueTimeMetricName, TimeUnit.MILLISECONDS, TimeUnit.SECONDS, brokerMetricTags(broker.id)
+      RequestRateAndQueueTimeMetricName, TimeUnit.MILLISECONDS, TimeUnit.SECONDS, brokerMetricTag
     )
+    timerMetricNameWithTag.computeIfAbsent(RequestRateAndQueueTimeMetricName, k => new java.util.HashSet[java.util.Map[String, String]]())
+      .add(brokerMetricTag)
 
     val requestThread = new RequestSendThread(config.brokerId, controllerEpoch, messageQueue, networkClient,
       brokerNode, config, time, requestRateAndQueueTimeMetrics, stateChangeLogger, threadName)
     requestThread.setDaemon(false)
 
-    val queueSizeGauge = metricsGroup.newGauge(QueueSizeMetricName, () => messageQueue.size, brokerMetricTags(broker.id))
+    val queueSizeGauge = metricsGroup.newGauge(QueueSizeMetricName, () => messageQueue.size, brokerMetricTag)
+    gaugeMetricNameWithTag.computeIfAbsent(QueueSizeMetricName, k => new java.util.HashSet[java.util.Map[String, String]]())
```

Comment:
these two lines are tightly coupled so we might want to create a method so that future implementors don't accidentally forget to call second line when writing a line similar to first.